modules/configmap/README.md

```diff
@@ -0,0 +1,102 @@
+# `configmap`
+
+This module encapsulates the creation of a Google Secret Manager secret to hold
+a piece of data in a manner that can be used as an environment variable or
+volume with Cloud Run. As the expectation is that this data is not sensitive it
+takes it directly to populate the secret.
+
+```hcl
+module "my-configmap" {
+  source = "chainguard-dev/common/infra//modules/configmap"
+
+  project_id = var.project_id
+  name       = "my-configmap"
+
+  # What the service accessing this configuration will run as.
+  service-account = google_service_account.foo.email
+
+  # The raw data of the configuration.
+  data = <<EOT
+  This is the data that will go into
+  the "configmap".
+  EOT
+
+  # Optionally: channels to notify if this configuration is manipulated.
+  notification-channels = [ ... ]
+}
+
+module "foo-service" {
+  source     = "chainguard-dev/common/infra//modules/regional-go-service"
+  project_id = var.project_id
+  name       = "foo"
+  regions    = module.networking.regional-networks
+
+  service_account = google_service_account.foo.email
+  containers = {
+    "foo" = {
+      source = {
+        working_dir = path.module
+        importpath  = "./cmd/foo"
+      }
+      ports = [{ container_port = 8080 }]
+      volume_mounts = [{
+        name       = "foo"
+        mount_path = "/var/run/foo/"
+      }]
+    }
+  }
+  volumes = [{
+    name = "foo"
+    secret = {
+      secret = module.my-configmap.secret_id
+      items = [{
+        version = module.my-configmap.version
```

Comment:
Do we want to encourage pinning to a particular version like this?

Reply:
I'd say there are trade-offs that folks need to think through.  By putting the version here, then changing the secret will trigger a new version, which will trigger a rollout.  By putting "latest" here, users can only pick up new versions if they use volumes and watch for changes.

As we saw today, this can have rollback implications at the service level.  Perhaps for `configmap` we have this use `deletion_policy = "ABANDON"` so that instead of deleting the older version we orphan it, then we can maybe have our cake and eat it too?

Reply:
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/secret_manager_secret_version#deletion_policy

Reply:
I like that, I'm going to do it 🤓